buildSrc/src/main/java/io/micronaut/guides/feature/GitProperties.java

```diff
@@ -0,0 +1,78 @@
+package io.micronaut.guides.feature;
+
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.io.IOUtils;
+import io.micronaut.core.io.ResourceLoader;
+import io.micronaut.starter.application.ApplicationType;
+import io.micronaut.starter.application.generator.GeneratorContext;
+import io.micronaut.starter.build.dependencies.PomDependencyVersionResolver;
+import io.micronaut.starter.build.gradle.GradlePlugin;
+import io.micronaut.starter.build.maven.MavenPlugin;
+import io.micronaut.starter.feature.Feature;
+
+import javax.inject.Singleton;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+
+@Singleton
+public class GitProperties implements Feature {
+
+    private final PomDependencyVersionResolver resolver;
+    private final ResourceLoader loader;
+
+    public GitProperties(PomDependencyVersionResolver resolver,
+                         ResourceLoader loader) {
+        this.resolver = resolver;
+        this.loader = loader;
+    }
+
+    @Override
+    @NonNull
+    public String getName() {
+        return "git-properties";
+    }
+
+    @Override
+    public void apply(GeneratorContext generatorContext) {
+        if (generatorContext.getBuildTool().isGradle()) {
+            generatorContext.addBuildPlugin(GradlePlugin.builder()
+                    .id("com.gorylenko.gradle-git-properties")
+                    .lookupArtifactId("gradle-git-properties")
+                    .build());
+            return;
+        }
+
+        String version = resolver.resolve("git-commit-id-plugin")
+                .orElseThrow(() -> new RuntimeException("git-commit-id-plugin not found in pom.xml"))
+                .getVersion();
+        String pluginXml = loadPluginXml().replace("@VERSION@", version);
```

Comment:
We typically do this simpler. Instead of loading the plugin template and replacing the version we use the `BuildProperties`. See:  

https://github.com/micronaut-projects/micronaut-starter/blob/2.5.x/starter-core/src/main/java/io/micronaut/starter/feature/function/azure/AbstractAzureFunction.java#L92-L98